File: 백준/Silver/21921. 블로그/블로그.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int x = Integer.parseInt(st.nextToken());

        int [] visitors = new int[n];
        String[] inputs = br.readLine().split(" ");
        for(int i = 0;i<n;i++){
            visitors[i] = Integer.parseInt(inputs[i]);
        }

        long currentSum =0;
        for(int i = 0;i<x;i++){
            currentSum+=visitors[i];
        }

        long maxSum = currentSum;
        int cnt = 1;
        
        for(int i =x;i<n;i++){
            currentSum = currentSum - visitors[i-x]+visitors[i];

            if(currentSum>maxSum){
                maxSum = currentSum;
                cnt =1;
            }else if(currentSum == maxSum){
                cnt ++;
            }
        }
        if(maxSum==0){
            System.out.println("SAD");
        }else{
            System.out.println(maxSum);
            System.out.println(cnt);
        }

    }
}

File: 백준/Silver/21921. 블로그/블로그.py
import sys
input = sys.stdin.readline

n,x = map(int,input().split())
visitors = list(map(int,input().split()))

current_sum = sum(visitors[:x])
max_sum = current_sum
cnt = 1

for i in range(x,n):
    current_sum = current_sum - visitors[i-x] + visitors[i]
    if current_sum>max_sum:
        max_sum = current_sum
        cnt =1
    elif current_sum == max_sum:
        cnt +=1

if max_sum == 0:
    print("SAD")
else:
    print(max_sum)
    print(cnt)
